tests/feedstock_tests/check_build_numbers.py

```diff
@@ -46,52 +46,56 @@ def main(arg_strings=None):
-                                                                  "number" : meta.meta['build']['number']}
-
-    utils.run_and_log("git checkout {}".format(pr_branch))
-    current_pr_build_numbers = dict()
-    for recipe in build_config_data["recipes"]:
-        metas = conda_build.api.render(recipe['path'],
-                                    config=config,
-                                    variants=variants[0],
-                                    bypass_env_check=True,
-                                    finalize=False)
-        for meta,_,_ in metas:
-            current_pr_build_numbers[meta.meta['package']['name']] = {"version" : meta.meta['package']['version'],
-                                                                      "number" : meta.meta['build']['number']}
-
-    print("Current PR Build Info:    ", current_pr_build_numbers)
-    print("Master Branch Build Info: ", master_build_numbers)
-
-    #No build numbers can go backwards without a version change.
-    for package in master_build_numbers:
-        if package in current_pr_build_numbers and current_pr_build_numbers[package]["version"] == master_build_numbers[package]["version"]:
-            assert int(current_pr_build_numbers[package]["number"]) >= int(master_build_numbers[package]["number"]), "If the version doesn't change, the build number can't be reduced."
-
-    #If packages are added or removed, don't require a version change
-    if set(master_build_numbers.keys()) != set(current_pr_build_numbers.keys()):
-        return
-
-    #At least one package needs to increase the build number or change the version.
-    checks = [current_pr_build_numbers[package]["version"] != master_build_numbers[package]["version"] or
-              int(current_pr_build_numbers[package]["number"]) > int(master_build_numbers[package]["number"])
-                  for package in master_build_numbers]
-    assert any(checks), "At least one package needs to increase the build number or change the version."
+    recipe_conda_build_config = build_feedstock.get_conda_build_config()
+    if recipe_conda_build_config:
+        config.variant_config_files.append(recipe_conda_build_config)
+    config.verbose = False
+    variant_build_results = dict()
+    for variant in variants:
+        utils.run_and_log("git checkout {}".format(default_branch))
+        master_build_numbers = dict()
+        for recipe in build_config_data["recipes"]:
+            metas = conda_build.api.render(recipe['path'],
+                                        config=config,
+                                        variants=variant,
+                                        bypass_env_check=True,
+                                        finalize=False)
+            for meta,_,_ in metas:
+                master_build_numbers[meta.meta['package']['name']] = {"version" : meta.meta['package']['version'],
+                                                                    "number" : meta.meta['build']['number']}
+
```

Comment:
Do you think we should create a function for this section as it is doing same thing for master branch as well as PR branch? I know this is not called from anywhere else. But I thought  having a function will just ensure we are retrieving same information from both branches, in case something changes in future.

Reply:
Rest looks good to me.

Reply:
Thanks, that's a good idea. I made this change.